engine/src/main/java/org/pentaho/di/trans/steps/csvinput/CsvInputHelper.java

```diff
@@ -0,0 +1,120 @@
+/*! ******************************************************************************
+ *
+ * Pentaho
+ *
+ * Copyright (C) 2024 by Hitachi Vantara, LLC : http://www.pentaho.com
+ *
+ * Use of this software is governed by the Business Source License included
+ * in the LICENSE.TXT file.
+ *
+ * Change Date: 2029-07-20
+ ******************************************************************************/
+
+package org.pentaho.di.trans.steps.csvinput;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.apache.commons.vfs2.FileObject;
+import org.json.simple.JSONObject;
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.logging.LogChannelInterface;
+import org.pentaho.di.core.vfs.KettleVFS;
+import org.pentaho.di.i18n.BaseMessages;
+import org.pentaho.di.trans.TransMeta;
+import org.pentaho.di.trans.step.BaseStepHelper;
+import org.pentaho.di.trans.step.StepMeta;
+import org.pentaho.di.trans.steps.common.CsvInputAwareMeta;
+import org.pentaho.di.trans.steps.common.CsvInputAwareStep;
+import org.pentaho.di.trans.steps.fileinput.text.BufferedInputStreamReader;
+import org.pentaho.di.trans.steps.fileinput.text.CsvFileImportProcessor;
+
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Helper class for Csv Input step actions, extracted from CsvInput.
+ * Handles only action methods related to metadata extraction.
+ */
+public class CsvInputHelper extends BaseStepHelper implements CsvInputAwareStep {
+
+  public static final String STEP_NAME = "stepName";
+  public static final String ERROR = "error";
+  private static final String GET_FIELDS = "getFields";
+  private TransMeta transMeta;
+  public CsvInputHelper() {
+    super();
+  }
+
+  public void setTransMeta(TransMeta transMeta) {
```

Comment:
Why we adding transmeta at class level scope as it is needed only in the method scope. 
Please refer to "plugins/streaming/impls/jms/src/main/java/org/pentaho/di/trans/step/jms/JmsConsumerMeta.java" meta class on how to pass meta and/or transmeta.